File: server/http/handler/operation.go
package handler

import (
	"net/http"
	"strconv"

	"gitee.com/openeuler/PilotGo-plugin-llmops/server/service/operation"
	"github.com/gin-gonic/gin"
)

var operationSrv = operation.GetOperationService()

type operationScriptReq struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	Content     string `json:"content"`
	UpdatedBy   string `json:"updated_by"`
}

func ListOperationScripts(c *gin.Context) {
	projectID, err := strconv.Atoi(c.Param("id"))
	if err != nil || projectID <= 0 {
		ResponseError(c, http.StatusBadRequest, "invalid project id")
		return
	}
	data, err := operationSrv.List(projectID)
	if err != nil {
		ResponseError(c, http.StatusInternalServerError, err.Error())
		return
	}
	Response(c, data)
}

func CreateOperationScript(c *gin.Context) {
	projectID, err := strconv.Atoi(c.Param("id"))
	if err != nil || projectID <= 0 {
		ResponseError(c, http.StatusBadRequest, "invalid project id")
		return
	}
	var req operationScriptReq
	if err := c.ShouldBindJSON(&req); err != nil || req.Name == "" {
		ResponseError(c, http.StatusBadRequest, "invalid request")
		return
	}
	if err := operationSrv.Create(projectID, req.Name, req.Description, req.Content, req.UpdatedBy); err != nil {
		ResponseError(c, http.StatusInternalServerError, err.Error())
		return
	}
	ResponseOK(c, "created")
}

func UpdateOperationScript(c *gin.Context) {
	id, err := strconv.ParseInt(c.Param("sid"), 10, 64)
	if err != nil || id <= 0 {
		ResponseError(c, http.StatusBadRequest, "invalid id")
		return
	}
	var req operationScriptReq
	if err := c.ShouldBindJSON(&req); err != nil || req.Name == "" {
		ResponseError(c, http.StatusBadRequest, "invalid request")
		return
	}
	if err := operationSrv.Update(id, req.Name, req.Description, req.Content, req.UpdatedBy); err != nil {
		ResponseError(c, http.StatusInternalServerError, err.Error())
		return
	}
	ResponseOK(c, "updated")
}

func DeleteOperationScript(c *gin.Context) {
	id, err := strconv.ParseInt(c.Param("sid"), 10, 64)
	if err != nil || id <= 0 {
		ResponseError(c, http.StatusBadRequest, "invalid id")
		return
	}
	if err := operationSrv.Delete(id); err != nil {
		ResponseError(c, http.StatusInternalServerError, err.Error())
		return
	}
	ResponseOK(c, "deleted")
}

File: web/src/apis/operation.ts
import { httpClient } from './request'

export interface OperationScript {
  id: number | string
  name: string
  description: string
  content?: string
  updatedBy: string
  updatedAt: string
}

export async function listOperationScripts(projectId: number | string, page?: number): Promise<OperationScript[]> {
  const res = await httpClient.get<any[]>(`/api/project/${projectId}/operation/scripts`, { page })
  const list = res.data ?? []
  // 后端下划线 → 前端驼峰映射
  return (Array.isArray(list) ? list : []).map((it: any) => ({
    id: it.id,
    name: it.name,
    description: it.description,
    content: it.content,
    updatedBy: it.updated_by ?? '',
    updatedAt: it.updated_at ?? '',
  }))
}

export async function deleteOperationScript(
  projectId: number | string,
  scriptId: number | string,
): Promise<string> {
  const res = await httpClient.delete<void>(`/api/project/${projectId}/operation/scripts/${scriptId}`)
  return res.message ?? ''
}

export interface OperationScriptPayload {
  name: string
  description?: string
  content?: string
  updatedBy?: string
}

export async function createOperationScript(
  projectId: number | string,
  payload: OperationScriptPayload,
): Promise<string> {
  // 前端驼峰 → 后端下划线
  const res = await httpClient.post<void>(`/api/project/${projectId}/operation/scripts`, {
    name: payload.name,
    description: payload.description,
    content: payload.content,
    updated_by: payload.updatedBy,
  })
  return res.message ?? ''
}

export async function updateOperationScript(
  projectId: number | string,
  scriptId: number | string,
  payload: OperationScriptPayload,
): Promise<string> {
  // 前端驼峰 → 后端下划线
  const res = await httpClient.put<void>(`/api/project/${projectId}/operation/scripts/${scriptId}`, {
    name: payload.name,
    description: payload.description,
    content: payload.content,
    updated_by: payload.updatedBy,
  })
  return res.message ?? ''
}
